tests/integration/ledger/ledger_test.go

```diff
@@ -0,0 +1,236 @@
+package ledger_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/keys"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/simapp/params"
+
+	"github.com/evmos/ethermint/crypto/ethsecp256k1"
+	"github.com/evmos/ethermint/encoding"
+	"github.com/evmos/evmos/v10/app"
+	"github.com/evmos/evmos/v10/tests/integration/ledger/mocks"
+	"github.com/evmos/evmos/v10/testutil"
+
+	"github.com/spf13/cobra"
+
+	sdktestutil "github.com/cosmos/cosmos-sdk/testutil"
+	sdktestutilcli "github.com/cosmos/cosmos-sdk/testutil/cli"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	bankcli "github.com/cosmos/cosmos-sdk/x/bank/client/cli"
+
+	. "github.com/onsi/ginkgo/v2"
+)
+
+var (
+	signOkMock = func(_ []uint32, msg []byte) ([]byte, error) {
+		return s.privKey.Sign(msg)
+	}
+
+	signErrMock = func(_ []uint32, msg []byte) ([]byte, error) {
+		return nil, mocks.ErrMockedSigning
+	}
+)
+
+var _ = Describe("ledger cli and keyring functionality", func() {
+	var (
+		receiverAccAddr sdk.AccAddress
+		encCfg          params.EncodingConfig
+		kr              keyring.Keyring
+		mockedIn        sdktestutil.BufferReader
+		clientCtx       client.Context
+		ctx             context.Context
+		cmd             *cobra.Command
+		krHome          string
+		keyRecord       *keyring.Record
+	)
+
+	ledgerKey := "ledger_key"
+
+	s.SetupTest()
+	s.SetupEvmosApp()
+
+	Describe("Perform key addition", func() {
```

Comment:
structure
The `describe`, `context` and `it` blocks should compose test cases that are easy to read, without having to understand the code. Can you change the structure to the following? 

```
var _ = Describe("Ledger CLI: ", func() {
	Describe("Adding a key from ledger using the CLI", func() {
		Context("with eth_secp256k1", func() {
			It("should add the ledger key to the keyring", func() {})
		})
		Context("with secp256k1", func() {
			It("should return an error on ledger", func() {})
		})
	})
	
	Describe("Singing a transaction with ledger using the CLI", func() {
		Context("before confirming the signature", func() {
			Context("with valid key in keyring", func() {
				It("should prompt confirmation to sign message", func() {})
			}
			Context("with invalid key in keyring", func() {
				It("should raise error from ledger signing", func() {})
			}
		}
		
		Context("after confirming the signature", func() {
			Context("with valid key in keyring", func() {
				It("should execute tx", func() {
			})
			Context("with invalid key in keyring", func() {
				It("should not execute tx", func() {
			})
		})
	})
})
```

Reply:
@danburck check updates please